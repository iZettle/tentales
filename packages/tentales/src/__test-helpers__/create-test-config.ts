/* tslint:disable no-empty */

export function createTestConfig() {
  return {
    port: 12345,
    public: true,
    auth: {
      serverSecret: "test-secret",
    },
    reactComponentsDirectory: "",
    services: {
      renderer: {
        host: "this",
        path: "/tt/renderer",
      },
      data: {
        host: "this",
        path: "/tt/data",
      },
      editor: {
        host: "this",
        path: "/tt/editor",
      },
    },
    hooks: {
      middlewares: {
        beforeRenderMiddleware: [() => {}, () => {}],
        rendererService: [() => {}],
      },
    },
  }
}
