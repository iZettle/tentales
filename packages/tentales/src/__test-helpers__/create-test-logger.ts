import { Log } from "tentales-log"

export const createTestLogger = (): Log => ({
  silly: jest.fn(),
  verbose: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
})
