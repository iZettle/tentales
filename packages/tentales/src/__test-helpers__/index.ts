export { createTestConfig } from "./create-test-config"
export { createTestMwConfig } from "./create-test-mw-config"
export { createTestMw } from "./create-test-mw"
