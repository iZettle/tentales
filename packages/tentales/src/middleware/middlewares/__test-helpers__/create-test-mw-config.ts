import { always } from "ramda"

export function createTestMwConfig() {
  const ttConfig = {
    port: 12345,
    public: true,
    auth: {
      serverSecret: "",
    },
    reactComponentsDirectory: "",
    services: {
      renderer: {
        host: "this",
        path: "/tt/renderer",
      },
      data: {
        host: "this",
        path: "/tt/data",
      },
      editor: {
        host: "this",
        path: "/tt/editor",
      },
    },
    hooks: {
      middlewares: {},
    },
  }

  return {
    services: {
      renderer: async () => await {},
      data: async () => await {},
      editor: async () => await {},
    },
    log: {
      silly: always(null),
      verbose: always(null),
      info: always(null),
      warn: always(null),
      error: always(null),
    },
    config: ttConfig,
  }
}
