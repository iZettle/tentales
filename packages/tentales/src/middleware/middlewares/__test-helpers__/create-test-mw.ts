/* tslint:disable no-any no-empty */

export function createTestMw({
  body = "Next body",
  contentType = null,
  status = null,
  fn = (ctx: any) => {},
} = {}) {
  return async function nextMw(ctx: any, next: any) {
    ctx.body = body

    const header = {}

    if (contentType) {
      ctx.response.type = contentType
    }

    if (status) {
      ctx.status = status
    }

    if (header) {
      ctx.set(header)
    }

    fn(ctx)

    await next()
  }
}
