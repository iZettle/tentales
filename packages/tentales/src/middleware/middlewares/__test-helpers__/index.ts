export { createTestMwConfig } from "./create-test-mw-config"
export { createTestMw } from "./create-test-mw"
