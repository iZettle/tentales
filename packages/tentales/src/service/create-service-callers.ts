import fetch from "node-fetch"
import { createLog } from "tentales-log"
import { Action, ServiceConfig, ServiceCaller, Config } from "tentales"

// tslint:disable-next-line:no-any
function getServiceConfig(serviceName: string, config: any): ServiceConfig {
  return config.services[serviceName] as ServiceConfig
}

export function createServiceCallers(config: Config): ServiceCaller[] {
  return Object.keys(config.services).map(serviceName => {
    const log = createLog("TT")
    const serviceConfig = getServiceConfig(serviceName, config)
    const func = async <T>(action: Action<T>) => {
      const actionString = JSON.stringify(action)
      const host =
        serviceConfig.host === "this"
          ? `http://localhost:${config.port}`
          : serviceConfig.host
      log.silly(`Calling service on ${host}${serviceConfig.path}`)
      // TODO, add timeout
      const response = await fetch(`${host}${serviceConfig.path}`, {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: actionString,
      })
      return await response.json()
    }

    return {
      func,
      name: serviceName,
      config: serviceConfig,
    }
  })
}
