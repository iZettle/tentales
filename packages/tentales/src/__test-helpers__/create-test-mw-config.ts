import { always } from "ramda"
import { createTestConfig } from "./create-test-config"

export function createTestMwConfig() {
  return {
    services: {
      renderer: async () => await {},
      data: async () => await {},
      editor: async () => await {},
    },
    log: {
      silly: always(null),
      verbose: always(null),
      info: always(null),
      warn: always(null),
      error: always(null),
    },
    config: createTestConfig(),
  }
}
