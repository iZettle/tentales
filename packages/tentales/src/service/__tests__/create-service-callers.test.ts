import { createServiceCallers } from "../create-service-callers"
import { createTestConfig } from "../../__test-helpers__"

test("It creates service callers", async () => {
  const resp = createServiceCallers(createTestConfig())
  expect(resp).toMatchSnapshot()
})
