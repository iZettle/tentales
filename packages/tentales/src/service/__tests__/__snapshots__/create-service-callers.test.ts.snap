// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It creates service callers 1`] = `
Array [
  Object {
    "config": Object {
      "host": "this",
      "path": "/tt/renderer",
    },
    "func": [Function],
    "name": "renderer",
  },
  Object {
    "config": Object {
      "host": "this",
      "path": "/tt/data",
    },
    "func": [Function],
    "name": "data",
  },
  Object {
    "config": Object {
      "host": "this",
      "path": "/tt/editor",
    },
    "func": [Function],
    "name": "editor",
  },
]
`;
